R/doeFactorial.R

```diff
@@ -267,15 +278,30 @@ doeFactorial <- function(jaspResults, dataset, options, ...){
                  options[["factorialBlocks"]]) *
         options[["factorialCornerReplicates"]] +
         options[["repeatRuns"]]
-      resolution <-
-        if(log2(as.numeric(options[["factorialRuns"]])) < options[["numberOfFactors"]]){
-          as.character(as.roman(DoE.base::design.info(
-            FrF2::FrF2(nfactors = options[["numberOfFactors"]],
-                       nruns = as.numeric(options[["factorialRuns"]]))
-            )$catlg.entry[[1]]$res))
-      } else {
-        "Full"
-      }
+
+      # Resolution based on SKF's diagram
+      sumResolution    <- runs + options[["numberOfFactors"]]
+      resolution.Full  <- c(6, 11, 20, 37, 70, 135)
+      resolution.six   <- c(38, 137)
+      resolution.Five  <- c(21, 72, 138, 139)
+      resolution.three <- c(7, 13, 14, 15, 25, 26, 27, 28, 29, 30, 31)
+
+
+      if (any(sumResolution == resolution.Full))
+        res <- "Full"
+      else if (any(sumResolution == resolution.six))
+        res <- "VI"
+      else if (any(sumResolution == resolution.Five))
+        res <- "V"
+      else if (any(sumResolution == resolution.three))
+        res <- "III"
+      else if (any(sumResolution == 137))
+        res <- "VIII"
+      else if (any(sumResolution == 71))
+        res <- "VII"
+      else
+        res <- "III"
```

Comment:
if `sumResolution` is a single number, this should be
```r
       if (sumResolution %in% resolution.Full)) 
```
same for the stuff below. If `sumResolution` is not a single number and the order matters, then what you did is fine.